emu_mps/optimatrix/permutations.py

```diff
@@ -1,6 +1,33 @@
 import torch
 
 
+def eye_permutation(n: int) -> torch.Tensor:
```

Comment:
So, I have an idea for a future refactor:
create a `Permutation` class with methods
```
class Permutation:
  def __init__(self, ...):
    # ...


  @classmethod
  def id(cls, n: int):
    return cls(torch.arange(n))

 
  def is_identity(self):
    return torch.equals(self.perm, torch.arange(len(self.perm)))


  def apply(self, input: list | str | torch.Tensor):
    if isinstance(input, list):
      return [input_list[i] for i in self.perm.tolist()]
    elif isinstance(input, str):
      return "".join(self.apply(list(input)))
    elif isinstance(input, torch.Tensor):
      # ...


  def inverse(self) -> Permutation:
    inv_perm = torch.empty_like(self.perm)
    inv_perm[self.perm] = np.arange(len(self.perm))
    return Permutation(inv_perm) 
```

Reply:
why not !